Sources/Sentry/SentryClient.m

```diff
@@ -144,9 +142,9 @@ - (instancetype)initWithOptions:(SentryOptions *)options
                  threadInspector:threadInspector
                           random:[SentryDependencyContainer sharedInstance].random
                     crashWrapper:[SentryCrashWrapper sharedInstance]
-                   deviceWrapper:deviceWrapper
                           locale:[NSLocale autoupdatingCurrentLocale]
-                        timezone:[NSCalendar autoupdatingCurrentCalendar].timeZone];
+                        timezone:[NSCalendar autoupdatingCurrentCalendar].timeZone
+            extraContextProvider:extraContextProvider];
```

Comment:
m: instead of requesting an `extraContextProvider` here, I think you should rollback the init functions, and initialize extraContextProvider inside the init, and provide it with the needed wrappers as mentioned in this comment, then you wont need to fix the tests.